lib/ecto/model/timestamps.ex

```diff
@@ -23,11 +23,23 @@ defmodule Ecto.Model.Timestamps do
     if get_change changeset, field do
       changeset
     else
-      {date, {h, m, s}} = :erlang.universaltime
-      put_change changeset, field, Ecto.Type.load!(type, {date, {h, m, s, 0}})
+      put_change changeset, field, Ecto.Type.load!(type, timestamp_tuple_with_usec(changeset.repo.adapter))
     end
   end
 
+  defp timestamp_tuple_with_usec(Ecto.Adapters.MySQL) do
```

Comment:
Ah, this is interesting. We shouldn't read the adapter configuration though because it is not cached and it is going to affect considerably the insertion/update times. We would need to think about an efficient way of doing this.